cmd/ocsp-responder/main.go

```diff
@@ -162,6 +167,21 @@ type dbSelector interface {
 
 // Response is called by the HTTP server to handle a new OCSP request.
 func (src *dbSource) Response(ctx context.Context, req *ocsp.Request) ([]byte, http.Header, error) {
+	// Metrics for response lookups
+	ocspLookups := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "ocsp_lookups",
+		Help: "A counter of ocsp lookups labeled by source_result",
+	}, []string{"result"})
+	src.metrics.MustRegister(ocspLookups)
+
+	sourceUsed := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "lookup_source_used",
+		Help: "A counter of lookups returned labeled by source used",
+	}, []string{"source"})
+	src.metrics.MustRegister(sourceUsed)
```

Comment:
You don't want to register these stats each time Response is called; you need to call them once when the dbSource in constructed, and save them on the dbSource object.

Reply:
Thanks! It seemed like the things I was passing around started getting big and with the added metrics I went ahead and created a responder object to pass the logger, clock, and any additional new metrics around easier.